desc/objectives/_omnigenity.py

```diff
@@ -540,6 +541,319 @@ def compute(self, params, constants=None):
+
+        self._dim_f = field_grid.num_nodes
+        self._eq_data_keys = ["|B|_mn"]
+        self._field_data_keys = ["theta_B", "zeta_B", "|B|_omni"]
+
+        errorif(
+            eq_grid.NFP != field_grid.NFP,
+            msg="eq_grid and field_grid must have the same number of field periods",
+        )
+        errorif(eq_grid.sym, msg="eq_grid must not be symmetric")
+        errorif(field_grid.sym, msg="field_grid must not be symmetric")
+        errorif(eq_grid.num_rho != 1, msg="eq_grid must be a single surface")
+        errorif(field_grid.num_rho != 1, msg="field_grid must be a single surface")
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        profiles = get_profiles(self._eq_data_keys, obj=eq, grid=eq_grid)
+        eq_transforms = get_transforms(
+            self._eq_data_keys,
+            obj=eq,
+            grid=eq_grid,
+            M_booz=M_booz,
+            N_booz=N_booz,
+        )
+        field_transforms = get_transforms(
+            self._field_data_keys,
+            obj=field,
+            grid=field_grid,
+        )
+
+        # compute returns points on the grid of the field
+        # (dim_f = field_grid.num_nodes)
+        # so set quad_weights to the surface grid
+        # to avoid it being incorrectly set in the super build
+        w = field_grid.weights
+        w *= jnp.sqrt(field_grid.num_nodes)
+
+        self._constants = {
+            "eq_profiles": profiles,
+            "eq_transforms": eq_transforms,
+            "field_transforms": field_transforms,
+            "quad_weights": w,
+            "helicity": self.helicity,
+        }
+
+        if self._eq_fixed:
+            # precompute the eq data since it is fixed during the optimization
+            eq_data = compute_fun(
+                "desc.equilibrium.equilibrium.Equilibrium",
+                self._eq_data_keys,
+                params=self._eq.params_dict,
+                transforms=self._constants["eq_transforms"],
+                profiles=self._constants["eq_profiles"],
+            )
+            self._constants["eq_data"] = eq_data
+        if self._field_fixed:
+            # precompute the field data since it is fixed during the optimization
+            field_data = compute_fun(
+                "desc.magnetic_fields.OmnigenousField",
+                self._field_data_keys,
+                params=self._field.params_dict,
+                transforms=self._constants["field_transforms"],
+                profiles={},
+                helicity=self._constants["helicity"],
+            )
+            self._constants["field_data"] = field_data
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        if self._normalize:
+            # average |B| on axis
+            self._normalization = jnp.mean(field.B_lm[: field.M_well])
```

Comment:
Can you guarantee that this always be positive? What if the B field is decreasing radially? if not, should throw an absolute value here

Reply:
added a check to make sure the normalization is positive